src/Services/ComposerConflictsRenovator.php

```diff
@@ -0,0 +1,258 @@
+
+            $updated_result = $this->updateConflictsForVulnerability($entry, $this->composer_json_content);
+            if ($updated_result?->isUpdated()) {
+                try {
+                    $new_composer_json_content = $updated_result->getComposerJsonContent();
+                    $commit_message            = sprintf(
+                        '%1$s %2$s | CVSS = %3$s | %4$s',
+                        $software_type,
+                        $software_name,
+                        $cvss,
+                        $updated_result->getConflictVersionsString() ?? '',
+                    );
+
+                    $encoded     = json_encode(
+                        value: $new_composer_json_content,
+                        flags: JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES
+                    );
+                    $branch_name = $entry['id'] ?? md5(json_encode($entry['software']));
+
+                    $this->github_api_controller->createBranch($branch_name, static::REPO_DEFAULT_BRANCH_NAME);
+                    $this->github_api_controller->updateFileContent(
+                        static::COMPOSER_JSON_PATH,
+                        $encoded,
+                        $commit_message,
+                        $this->github_api_controller->getFileSha(
+                            static::COMPOSER_JSON_PATH,
+                            static::REPO_DEFAULT_BRANCH_NAME
+                        ),
+                        $branch_name
+                    );
+                    $this->github_api_controller->createPullRequest(
+                        static::REPO_DEFAULT_BRANCH_NAME,
+                        $branch_name,
+                        $commit_message,
+                        sprintf(
+                            "According to [Wordfence](https://www.wordfence.com/threat-intel/vulnerabilities/), %1\$s %2\$s has a %3\$s CVSS security vulnerability\n\nI'm bumping versions to %4\$s\n\nReferences: %5\$s",
+                            $software_type,
+                            $software_name,
+                            $cvss,
+                            $updated_result->getConflictVersionsString() ?? '',
+                            implode(' , ', $software[0]['references'] ?? [])
+                        )
+                    );
+                } catch (Exception $e) {
+                    continue;
+                }
+            }
+
+            sleep($pause);
+        }
+    }
+
+    /**
+     * Updates composer.json conflict section for specified vulnerability.
+     *
+     * @param array $entry
+     * @param array $composer_json_content
+     *
+     * @return ConflictSectionUpdateResult|null
+     */
+    protected function updateConflictsForVulnerability(
+        array $entry,
+        array $composer_json_content
+    ): ?ConflictSectionUpdateResult {
+        $updated  = false;
+        $software = $entry['software'];
+        if (empty($software)) {
+            return null;
+        }
```

Comment:
probably not necessary since you already verified before in the caller.

Reply:
Well, the function itself doesn't know that, so it should also check for it